lib/decompress/huf_decompress.c

```diff
@@ -263,7 +264,7 @@ static size_t HUF_initRemainingDStream(BIT_DStream_t* bit, HUF_DecompressAsmArgs
 
     /* Construct the BIT_DStream_t. */
     bit->bitContainer = MEM_readLE64(args->ip[stream]);
-    bit->bitsConsumed = ZSTD_countTrailingZeros((size_t)args->bits[stream]);
+    bit->bitsConsumed = ZSTD_countTrailingZeros64(args->bits[stream]);
```

Comment:
I note that `bitContainer` is of type `size_t`,
so it seems to me that the issue is the previous `MEM_readLE64()`.

I presume a `-Wconversion` warning flag would have caught such silent downsizing.

Reply:
Update from Nick: this code only runs on x86-64. So my change is safe.

Reply:
> Update from Nick: this code only runs on x86-64. So my change is safe.

It works, yes.
But as many things, it works fine today, and thanks to complex environmental guarantees, that are easy to forget.
Such knowledge can be lost in time, and some future committer may inadvertently trigger a bug there.
So part of the job is to make such future mistake more difficult to happen.

I can imagine 2 ways : 
- Stay on `(size_t)`, because it's the official type for this field. It happens to be 64-bit, but that's secondary (the code would still be technically correct for 32-bit too). You can leave a comment about the field width to improve understanding of future reviewers.
- Move firmly into `64-bit`, but in this case, `STATIC_ASSERT` that `sizeof(size_t)==8`. This will ensure that this code cannot be compiled in 32-bit mode, where this assumption would be wrong. It also justifies why a 64-bit type is used here instead of the expected `size_t`.

Note that some compiler might still not like the `U64 == size_t` equivalence, even when they are both 64-bit, on the ground that they are different types. As usual for this type of warnings, it depends. Compilers can change their mind, even between versions.